src/log.js

```diff
@@ -188,6 +189,20 @@ export class Log {
           Array.prototype.slice.call(arguments, 1));
       this.prepareError_(error);
       this.win.setTimeout(() => {throw error;});
+      return error;
+    }
+  }
+
+  /**
+   * Reports an error message and marks with an expected property. If the
+   * logging is disabled, the error is rethrown asynchronously.
+   * @param {string} tag
+   * @param {...*} var_args
+   */
+  expectedError(tag, var_args) {
+    const error = this.error(tag, var_args);
+    if (error) {
+      error.expected = true;
```

Comment:
While correct, it's a bit weird to set flag on an error after it's been reported. Maybe better refactor call into a private method and call from both methods with different flags?